### PR TITLE
VAGOV-3202 | VAGOV-3978 linkit implementation

### PR DESCRIPTION
**Pull request by ethanteague**
## What this does
- Sets linkit as the default link creator in wysiwyg and in link fields

## How to test
 - Go to `/node/715/edit`
 - In wyswyg field, select link icon, and begin typing - you should see a linkit dropdown
 - Add `List of link teasers` paragrpah
 - Begin begin typing in link field - you should see a linkit dropdown

## Definition of done
 - Above testing steps are successful

## Screenshots
linkit-form
should-show-drop-wysi
shoudl-drop-field

### CODE REVIEW

**Comment by ElijahLynn** [comment]
There is no linkit module in composer.json here. This commit is empty https://github.com/department-of-veterans-affairs/va.gov-cms/pull/502/commits/8e6ae08481ed0b781d21fed877ae838786b31084 (Adding linkit module and patches.)

**Comment by kevwalsh** [comment]
@ethanteague This is awesome!  I just added a commit to your branch to improve the "metadata" that appears in linkit `[node:field_administration] | Created by [node:author] on [node:created:html_date]`

**Comment by ethanteague** [comment]
> @ethanteague This is awesome! I just added a commit to your branch to improve the "metadata" that appears in linkit `[node:field_administration] | Created by [node:author] on [node:created:html_date]`

Thanks for doing that, @kevwalsh :+1: - I forgot to update that in my mr after we discussed it on our zoom call.

File: config/sync/views.view.content.yml [hunk 1]
```diff
@@ -36,7 +37,14 @@ display:
         type: role
         options:
           role:
+            content_api_consumer: content_api_consumer
```
Comment (kevwalsh) [inline config/sync/views.view.content.yml]: This file should not have changed as part of this work, looks like you overwrote some recnetly changed config.

File: config/sync/views.view.media.yml [hunk 2]
```diff
@@ -12,15 +12,13 @@ dependencies:
     - media.type.image
     - system.menu.admin
     - taxonomy.vocabulary.administration
-    - user.role.administrator
```
Comment (kevwalsh) [inline config/sync/views.view.media.yml]: This too should not have changed.

**Comment by ethanteague** [comment]
> There is no linkit module in composer.json here. This commit is empty [8e6ae08](https://github.com/department-of-veterans-affairs/va.gov-cms/commit/8e6ae08481ed0b781d21fed877ae838786b31084) (Adding linkit module and patches.)

Updated to inlcude.

File: config/sync/views.view.content.yml (hunk 1 above)
Comment (ethanteague) [reply config/sync/views.view.content.yml]: Updated, @kevwalsh

File: config/sync/views.view.media.yml (hunk 2 above)
Comment (ethanteague) [reply config/sync/views.view.media.yml]: Updated, @kevwalsh

File: composer.lock [hunk 3]
```diff
@@ -2521,7 +2517,7 @@
             "description": "Provides functionality for storing, validating and displaying international postal addresses.",
             "homepage": "http://drupal.org/project/address",
             "support": {
-                "source": "https://git.drupalcode.org/project/address"
+                "source": "http://cgit.drupalcode.org/address"
```
Comment (swirtSJW) [inline composer.lock]: Something interesting going on here.  Kev's commits are toggling the sources to git.drupalcode (the new format) and your commits toggle things back to cgit.drupalcode.   Uncertain of the cause at this point.  Not sure it is a big issue but is an interesting pissing contest that seems to happen  as you each make commits.  ;)

**Comment by swirtSJW** [comment]
I don't have a way to test this so I'd like @kevwalsh to pull it down and test it before merging.

**Comment by ethanteague** [comment]
> I notice there is a perms change on docroot/sites/default/settings.devshop.php Was that expected?
> If you think it is required, I'd like Jon P to weigh in on it.

I noticed with my last rebase against develop that the perms changed, @swirtSJW - let's ask @jonpugh if this is legit. Hey, @jonpugh - is this legit?

File: composer.lock (hunk 3 above)
Comment (ethanteague) [reply composer.lock]: Ugh.....this doesn't make _any_ sense. I would expect something like this to occur _only_ after a change being made to composer.json, @swirtSJW, but @kevwalsh didn't touch said file. Any ideas?

**Comment by ethanteague** [comment]
> I don't have a way to test this so I'd like @kevwalsh to pull it down and test it before merging.

:+1: - we also have the above behat issue that @ElijahLynn is trying to debug.

File: composer.lock (hunk 3 above)
Comment (jonpugh) [reply composer.lock]: Possibly a local git caching issue?  

Try `rm -rf ~/.composer` and be patient on the next composer install.

**Comment by ElijahLynn** [comment]
> > I notice there is a perms change on docroot/sites/default/settings.devshop.php Was that expected?
> > If you think it is required, I'd like Jon P to weigh in on it.
> 
> I noticed with my last rebase against develop that the perms changed, @swirtSJW - let's ask @jonpugh if this is legit. Hey, @jonpugh - is this legit?

Perms for all settings files should be the same, there shouldn't be any changes to it in this PR.

**Comment by jonpugh** [comment]
> > I notice there is a perms change on docroot/sites/default/settings.devshop.php Was that expected?
> > If you think it is required, I'd like Jon P to weigh in on it.
> 
> I noticed with my last rebase against develop that the perms changed, @swirtSJW - let's ask @jonpugh if this is legit. Hey, @jonpugh - is this legit?

Good catch!

@ethanteague I traced it to a commit you made: https://github.com/department-of-veterans-affairs/va.gov-cms/commit/6b73c744b9a4b41264c7e32d0acb686c6b05ed13

You can click the "..." next to the file, and click "View File" and you will see the file at the commit. 

Can you revert?

Permissions on that file don't actually matter in the repo, but we should not change them. 

Make sure to avoid committing any more than you have to, in general. Thanks!

**Comment by ElijahLynn** [comment]
> > I don't have a way to test this so I'd like @kevwalsh to pull it down and test it before merging.
> 
> @swirtSJW  - we also have the above behat issue that @ElijahLynn is trying to debug.

Passes tests locally for me. Working on Behat issue to pass CI.

**Comment by ethanteague** [comment]
> Permissions on that file don't actually matter in the repo, but we should not change them.
> 
> Make sure to avoid committing any more than you have to, in general. Thanks!

My bad, @jonpugh - I remember changing the perms so I could get a rebase unlocked (it was giving me permission denied errors when I tried to do any git operation). Generally I wouldn't touch anything not belonging to me - something about this file blew up my dev, and I wasn't able to get this particular file to behave without doing everything via `sudo`.

File: composer.lock (hunk 3 above)
Comment (ethanteague) [reply composer.lock]: Removing ~/.composer makes me nervous, @jonpugh. Should I just be targeting ~/.composer/cache?

**Comment by jonpugh** [comment]
@ethanteague @kevwalsh Bizarre behat behavior is fixed: Tests are passing.

Do not add any behat tests until I merge pr #493 , hopefully in the next few hours

**Comment by ethanteague** [comment]
> @ethanteague @kevwalsh Bizarre behat behavior is fixed: Tests are passing.
> 
> Do not add any behat tests until I merge pr #493 , hopefully in the next few hours

:+1:

**Comment by ElijahLynn** [comment]
This PR is good to go, this PR isn't adding any Behat tests and is passing all existing tests now.

File: composer.lock (hunk 3 above)
Comment (swirtSJW) [reply composer.lock]: @ethanteague  Just for laughs:  (really creepy nerdy laughs)
```
composer clear-cache
composer update --lock
```

Comment (swirtSJW) [reply composer.lock]: or perhaps 
```
composer clear-cache
composer install
composer update --lock
```

Comment (ethanteague) [reply composer.lock]: Both methods give me this diff, @swirtSJW ```VAGOV-3202_3978-linkit-implementation-et└─ $ git diff
diff --git a/composer.lock b/composer.lock
index fe49ff1e..b3d0df4d 100644
--- a/composer.lock
+++ b/composer.lock
@@ -2941,7 +2941,7 @@
             "version": "8.2.12",
             "source": {
                 "type": "git",
-                "url": "https://git.drupal.org/project/coder.git",
+                "url": "https://git.drupalcode.org/project/coder.git",
                 "reference": "984c54a7b1e8f27ff1c32348df69712afd86b17f"
             },
             "require": {
``` think I should update the lock file manually?

Comment (ethanteague) [reply composer.lock]: Actually....updating manually is probably not a good idea.

**Pull request merged**